src/txn/txn_timestamp.c

```diff
@@ -369,12 +369,22 @@ __wt_txn_global_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     last_oldest_ts = txn_global->oldest_timestamp;
     last_stable_ts = txn_global->stable_timestamp;
 
-    /* It is a no-op to set the oldest or stable timestamps behind the global values. */
-    if (has_oldest && txn_global->has_oldest_timestamp && oldest_ts <= last_oldest_ts)
-        has_oldest = false;
+    /* It is an invalid call to set the oldest or stable timestamps behind the global values. */
+    if (has_oldest && txn_global->has_oldest_timestamp && oldest_ts < last_oldest_ts) {
```

Comment:
@bvpvamsikrishna suggested having a stronger/more obvious message to help with any incoming failures from that code path since it's an API change. The message would be set back to what I am currently suggesting after a few months.
What are your thoughts?